flink-table/flink-table-common/src/main/java/org/apache/flink/table/connector/sink/abilities/SupportsTargetColumnWriting.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.connector.sink.abilities;
+
+import org.apache.flink.annotation.PublicEvolving;
+import org.apache.flink.table.connector.sink.DynamicTableSink;
+
+/**
+ * Interface for {@link DynamicTableSink}s that support target column writing.
+ *
+ * <p>The planner will parse target columns from the DML clause and call {@link
+ * #applyTargetColumns(int[][])} to pass an array of column index paths to sink.
+ *
+ * <p>The array indices are 0-based and support composite columns within (possibly nested)
+ * structures. This information comes from the column list of the DML clause, e.g., for a sink table
+ * t1 which schema is: {@code a STRING, b ROW < b1 INT, b2 STRING>, c BIGINT}
+ *
+ * <ul>
+ *   <li>insert: 'insert into t1(a, b.b2) ...', the column list will be 'a, b.b2', and will provide
```

Comment:
I like all the detail in the comment :-) 

I find this entry in the list confusing. what does the sentence "The statement 'insert into t1 select ...' will not apply this ability."

I assume the ability is whether it supports target column writing, which means when this interface is implemented. It seems there are cases when the interface is implemented, but the ability (target column writing) is not applied. It would be good to show examples where the ability is applied and examples where it is not. This list entry says the ability is not applied for an insert, what is the reason behind this?

Have I understood this correctly?

Reply:
Hi @davidradl , I've changed the comments to `The statement 'insert into t1 select ...' will provide an empty list.`